### PR TITLE
Aggregate the output for the command from multiple istiod

### PR DESCRIPTION
**Pull request by zhlsunshine**
This PR is for [issue#31703](https://github.com/istio/istio/issues/31703), and handle for multiple istiods debug information

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: istioctl/cmd/internal-debug.go [hunk 1]
```diff
@@ -139,57 +139,74 @@ By default it will use the default serviceAccount from (istio-system) namespace
...
+			svc, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), istiodServiceName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("please specify %q as %v", "--xds-address", err)
+			}
+			istiodNS, selector, serr := polymorphichelpers.SelectorsForObject(svc)
+			if serr != nil {
+				return fmt.Errorf("please specify %q as we cannot attach to %T: %v", "--xds-address", svc, serr)
+			}
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+			podList, plerr := kubeClient.CoreV1().Pods(istiodNS).List(context.TODO(), options)
```
Comment (irisdingbj) [inline istioctl/cmd/internal-debug.go:151]: You can use `kubeClient.GetIstioPods` directly

Comment (zhlsunshine) [reply istioctl/cmd/internal-debug.go:151]: @irisdingbj Okay, thanks Iris, I will check it!

**Comment by zhlsunshine** [comment]
Hi @irisdingbj  done, `kubeClient.GetIstioPods` works fine now

File: istioctl/cmd/internal-debug.go [hunk 2]
```diff
@@ -139,57 +138,80 @@ By default it will use the default serviceAccount from (istio-system) namespace
...
-				for _, v := range svc.Spec.Ports {
-					if v.Name == xdsPortName {
-						podPort = v.TargetPort.IntValue()
+			var isNull bool = false
+			var printErr error
+			// Iterate all istiod pods for retrieving debug information
+			for _, pod := range podList {
+				fmt.Println("")
+				fmt.Println("------------------------------------------------------------------------------")
+				fmt.Println("--- istioctl x debug for pod: ", pod.Name)
```
Comment (irisdingbj) [inline istioctl/cmd/internal-debug.go:173]: please add the namespace info as well.

Comment (zhlsunshine) [reply istioctl/cmd/internal-debug.go:173]: okay, it's very necessary to add the namespace of istiod pod. I will fix it and remove `---`. Thanks!

**Review by irisdingbj — APPROVED** [review]
Thanks!

**Pull request merged**